### PR TITLE
MultiServer: allow underscore as space replacement

### PR DESCRIPTION
**Pull request by Berserker66**
## What is this fixing or adding?
requests such as https://discord.com/channels/731205301247803413/1238036062346149909

## How was this tested?
with a few commands, such as 
```
/alias Test_potato Berserker
Named Test Potato as Berserker
/send tEsT_potato Progressive Dots
Notice (all): Cheat console: sending "Progressive Dots" to Berserker (Test Potato)
```

### CODE REVIEW

**Comment by PinkSwitch** [comment]
Will this break if a user has an actual underscore in their name? What about if there are 2 players: test_potato and test potato ?

**Comment by ScipioWright** [comment]
> Will this break if a user has an actual underscore in their name? What about if there are 2 players: test_potato and test potato ?

I think I would be fine with not allowing two players to have the same name where the only difference is one has an underscore where the other has a space

**Comment by Berserker66** [comment]
This is already addressed, you cannot have this.

**Comment by black-sliver** [comment]
You think this is better than trying to fix clients and command parsing to properly allow both spaces and underscores everywhere? Encapsulating string with spaces in `"` quotes is something even less tech-savvy people understand as far is i know. And we can provide command line documentation/recommendations in Network API so individual clients behave similar.

I won't block this, but I'll let the community decide (i.e. wait for enough peer review approvals before I hit approve).

**Comment by Berserker66** [comment]
This is a relatively simple and easy change we can  do now, a proper  command parser that understands  "  everywhere I'd say is still  desirable and may eventually replace this.

File: test/programs/test_multi_server.py
Comment (black-sliver) [inline test/programs/test_multi_server.py:20]: Note sure if you want to fix this, but this should probably be a separate test for readability.

File: MultiServer.py
Comment (black-sliver) [inline MultiServer.py:1956]:
```suggestion
        returns (team: int, slot: int, player name: str) if player was found or None otherwise
```

File: Generate.py
Comment (black-sliver) [inline Generate.py:213]: I'm wondering if it would be cheaper to do `" " -> "_"` rather than `"_" -> " "` because during runtime we will see more `"_"` than `" "`, but it shouldn't matter much.

File: MultiServer.py
Comment (agilbert1412) [inline MultiServer.py:1962]: The specific segment `x.lower().replace("_", " ")` is used in enough places that I think it would be worth centralizing it as a small method.

Then, if it ever happens that it is inverted, like Black Sliver suggested above, only place place needs to be changed

File: test/programs/test_multi_server.py
Comment (agilbert1412) [inline test/programs/test_multi_server.py:15]: Should we add the opposite test as well?

**Review by agilbert1412 — APPROVED** [review]
I think this is a good feature, and my comments are both very minor

**Review by Exempt-Medic — APPROVED** [review]
Briefly looked at the code but mostly did a lot of functional testing with various configurations of names to see that they would all work. I agree with Kaito's comments

File: test/programs/test_multi_server.py
Comment (beauxq) [inline test/programs/test_multi_server.py:12]: Can player name have underscore?

If so what happens if there's a player named `"a bc"` and another player named `"a_bc"`?

Comment (Exempt-Medic) [reply test/programs/test_multi_server.py:12]: https://github.com/ArchipelagoMW/Archipelago/pull/3283#issuecomment-2103772821

File: MultiServer.py
Comment (black-sliver) [inline MultiServer.py:1954]: We should disable loading of old multidata now?

File: MultiServer.py
Comment (black-sliver) [inline MultiServer.py:1962]: What happens if you load a multidata that was generated before this commit and has conflicting names? The check in Generate.py would have to be added before we actually allow using the new code, no?
(I.e. this would now make it impossible to use commands for the 2nd player if they are called "a_b" and "a b").

**Pull request closed without merging**
